aioriak/transport.py

```diff
@@ -150,6 +151,43 @@ def tail(self):
         return code, pbo
 
 
+class MapRedStream:
+    """
+    Wrapper for returning streaming result of MapReduce operation to user
+    """
+    def __init__(self, stream_parser, expect=None):
+        """
+        :param stream_parser: instance of StreamParser
+        """
+        self._stream_parser = stream_parser
+        self._buf = iter([])  # initialize with empty iterator
+        self._phase = None
+        self._expect = expect
+
+    async def __aiter__(self):
+        return self
+
+    async def __anext__(self):
+        try:
+            return self._phase, next(self._buf)
+        except StopIteration:
+            msg_code, pbo = await self._stream_parser.__anext__()
+            if self._expect and self._expect != msg_code:
+                raise Exception('Unexpected response code ({})'.format(msg_code))
```

Comment:
bc Transport._stream raise the same exception, I was just avoiding inconsistency